### PR TITLE
[New Rule] Suspicious WerFault Child Process

### PR DESCRIPTION
**Pull request by Samirbous**
## Issues

Resolves https://github.com/elastic/detection-rules/issues/211

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+creation_date = "2020/08/24"
+ecs_version = ["1.5.0"]
+maturity = "production"
+updated_date = "2020/08/24"
+
+[rule]
+author = ["Elastic"]
+description = """
+A suspicious WerFault child process was detected. This behavior may indicate an attempt to run unnoticed. Verify process
+details such as commandline, network connections, file writes and parent process details as well.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:11]:
```suggestion
A suspicious WerFault child process was detected, which may indicate an attempt to run unnoticed. Verify process
details such as command line, network connections, file writes and parent process details as well.
```

Will need to re-lint

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+[metadata]
+creation_date = "2020/08/24"
+ecs_version = ["1.5.0"]
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:3]:
```suggestion
ecs_version = ["1.6.0"]
```

Can accept change once #220 merges

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+risk_score = 47
+rule_id = "ac5012b8-8da8-440b-aaaf-aedafdea2dff"
+severity = "medium"
+tags = ["Elastic", "Windows"]
+type = "query"
+
+query = '''
+event.category :"process" and not event.type:"end" and
+ process.parent.name:"WerFault.exe"  and
+ not process.pe.original_file_name:(cofire.exe or psr.exe or VsJITDebugger.exe or TTTracer.exe or RUNDLL32.EXE)
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:31]: nit:

```suggestion
event.category:process and not event.type:end and
 process.parent.name:WerFault.exe  and
 not process.pe.original_file_name:(cofire.exe or psr.exe or VsJITDebugger.exe or TTTracer.exe or RUNDLL32.EXE)
```

Comment (brokensound77) [reply rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:31]: As of 7.8.0, winlogbeat did not use the `process.pe.original_file_name` field. Will need to verify if it is being used as of 7.9.0 or else use the field `winlog.event_data.OriginalFileName` for winlog compatibility

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+details such as commandline, network connections, file writes and parent process details as well.
+"""
+false_positives = ["Custom Windows Error Reporting Debugger"]
+index = ["winlogbeat-*", "logs-endpoint.events.*"]
+language = "kuery"
+license = "Elastic License"
+name = "Suspicious WerFault Child Process"
+references = [
+    "https://www.hexacorn.com/blog/2019/09/19/silentprocessexit-quick-look-under-the-hood/",
+    "https://github.com/sbousseaden/EVTX-ATTACK-SAMPLES/blob/master/Persistence/persistence_SilentProcessExit_ImageHijack_sysmon_13_1.evtx",
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:20]: 🎉

**Review by devonakerr — APPROVED** [review]

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+]
+risk_score = 47
+rule_id = "ac5012b8-8da8-440b-aaaf-aedafdea2dff"
+severity = "medium"
+tags = ["Elastic", "Windows"]
+type = "query"
+
+query = '''
+event.category:process and not event.type:end and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:29]:
```suggestion
event.category:process and event.type:(start or process_started) and
```

File: rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+ process.parent.name:WerFault.exe  and
```
Comment (brokensound77) [inline rules/windows/defense_evasion_masquerading_suspicious_werfault_childproc.toml:30]:
```suggestion
 process.parent.name:WerFault.exe and
```

**Review by brokensound77 — APPROVED** [review]
a few comments, then LGTM

**Pull request merged**